crates/egui-winit/src/lib.rs

```diff
@@ -865,70 +865,85 @@ impl State {
     }
 }
 
-/// Update the given viewport info with the current state of the window.
-///
-/// Call before [`State::take_egui_input`].
-pub fn update_viewport_info(
-    viewport_info: &mut ViewportInfo,
-    egui_ctx: &egui::Context,
-    window: &Window,
-) {
-    crate::profile_function!();
+pub fn math_inner_rect(window: &Window, pixels_per_point: Option<f32>) -> Option<Rect> {
+    let pixels_per_point = match pixels_per_point {
+        Some(pixels_per_point) => pixels_per_point,
+        None => window.scale_factor() as f32,
+    };
 
-    let pixels_per_point = pixels_per_point(egui_ctx, window);
+    let inner_pos_px = window
+        .inner_position()
+        .map(|pos| egui::emath::Pos2::new(pos.x as f32, pos.y as f32))
+        .ok();
 
-    let has_a_position = match window.is_minimized() {
-        None | Some(true) => false,
-        Some(false) => true,
+    let inner_size_px = {
+        let size = window.inner_size();
+        Some(egui::Vec2::new(size.width as f32, size.height as f32))
     };
 
-    let inner_pos_px = if has_a_position {
-        window
-            .inner_position()
-            .map(|pos| Pos2::new(pos.x as f32, pos.y as f32))
-            .ok()
+    let inner_rect_px = if let (Some(pos), Some(size)) = (inner_pos_px, inner_size_px) {
+        Some(egui::Rect::from_min_size(pos, size))
     } else {
         None
     };
 
-    let outer_pos_px = if has_a_position {
-        window
-            .outer_position()
-            .map(|pos| Pos2::new(pos.x as f32, pos.y as f32))
-            .ok()
-    } else {
-        None
-    };
+    inner_rect_px.map(|r| r / pixels_per_point)
+}
```

Comment:
You can simplify this code by early-outing with `?` (it works on `Option` too)

Reply:
> You can simplify this code by early-outing with `?` (it works on `Option` too)

This is the method I mainly use for readability, so please have emilk edit this part if necessary.